src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -475,6 +487,20 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
                 null;
         }
 
+        /// <summary>
+        /// Gets a path from a dot-source symbol.
+        /// </summary>
+        /// <param name="symbol">The symbol representing the dot-source expression.</param>
+        /// <param name="workspace">The current workspace</param>
+        /// <param name="scriptFile">The script file containing the symbol</param>
+        /// <returns></returns>
+        private static string GetDotSourcedPath(SymbolReference symbol, Workspace workspace, ScriptFile scriptFile)
+        {
+            string cleanedUpSymbol = PathUtils.NormalizePathSeparators(symbol.SymbolName.Trim('\'', '"'));
+            return workspace.ResolveRelativeScriptPath(Path.GetDirectoryName(scriptFile.FilePath),
+                Regex.Replace(cleanedUpSymbol, @"\$PSScriptRoot", Path.GetDirectoryName(scriptFile.FilePath), RegexOptions.IgnoreCase));
```

Comment:
Does `cleanedUpSymbol.Replace("$PSScriptRoot", Path.GetDirectoryName(scriptFile.FilePath))` not work here?

Reply:
It might be worth tweaking the regex a bit - @"\$PSScriptRoot|\${PSScriptRoot}" - to account for someone specifying this:
```
. "${PSScriptRoot}\foo.ps1"
```

Reply:
Bugger, that does look like the best option then. I think I have just been trained to hate regex